### PR TITLE
Hotfix #3323: Add in-progress check on release v1.0.34.2

### PR DESCRIPTION
**Pull request by mgulter**
https://github.com/endpointlabs/twrap-go/issues/3323

### CODE REVIEW

File: client/projects.go [hunk 1]
```diff
@@ -195,7 +205,40 @@ func (c *Client) ReleaseStatus(project, branch string) (*ReleaseStatus, error) {
...
+	for {
+		select {
+		case <-timeout:
+			return nil, fmt.Errorf("timeout [5min] exceeded while waiting for release status")
+		default:
+			if rs.ProgressStatus != ReleaseStatusHistoryInprogress {
+				return rs, nil
+			}
+
+			klog.Debugf("Release is still in progress for project [%s] on branch [%s]. Waiting for 5 seconds...", project, branch)
```
Comment (yeyisan) [inline client/projects.go:233]:
```suggestion
			klog.Debugf("Release status is still in progress for project [%s] on branch [%s]. Waiting for 5 seconds...", project, branch)
```

File: client/projects.go [hunk 2]
```diff
@@ -195,7 +205,40 @@ func (c *Client) ReleaseStatus(project, branch string) (*ReleaseStatus, error) {
...
+		return rs, nil
+	}
+
+	var opt = opts[0]
+	if !opt.WaitTillComplete {
+		return rs, nil
+	}
+
+	var err error
+	var timeout = time.After(opt.TotalWaitDurationToTimeout)
```
Comment (gormelof) [inline client/projects.go:222]:
```suggestion
	if len(opts) == 0 || !opts[0].WaitTillComplete {
		return rs, nil
	}

	opt := opts[0]
	timeout := time.After(opt.TotalWaitDurationToTimeout)
```

File: client/projects.go [hunk 3]
```diff
@@ -195,7 +205,40 @@ func (c *Client) ReleaseStatus(project, branch string) (*ReleaseStatus, error) {
...
+		case <-timeout:
+			return nil, fmt.Errorf("timeout [%s] exceeded while waiting for release status", opt.TotalWaitDurationToTimeout)
+		default:
+			if rs.ProgressStatus != ReleaseStatusHistoryInprogress {
+				return rs, nil
+			}
+
+			klog.Debugf("Release status is still in progress for project [%s] on branch [%s]. Waiting for 5 seconds...", project, branch)
+			time.Sleep(time.Second * 5)
+
```
Comment (gormelof) [inline client/projects.go:235]:
```suggestion
	var err error
```

**Review by gormelof — APPROVED** [review]
Looks good! I added some suggestions.

**Pull request closed without merging**